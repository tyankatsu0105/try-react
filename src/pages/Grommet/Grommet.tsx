import React, { FunctionComponent, useState } from 'react';
import { Helmet } from 'react-helmet';
import { Box, Button, Stack, Text } from 'grommet';
import { Notification, Chrome } from 'grommet-icons';
import styled from '@emotion/styled';

const Grommet: FunctionComponent = () => {
  const [count, setCount] = useState(0);

  const increase = () => {
    setCount(count + 1);
  };

  const Rotate = styled.div`
    display: flex;
    transform-origin: center;
    animation: spin 1.5s linear infinite;

    @keyframes spin {
      0% {
        transform: rotate(0deg);
      }
      100% {
        transform: rotate(360deg);
      }
    }
  `;
  return (
    <>
      <Helmet>
        <title>This is Grommet</title>
      </Helmet>

      <Box
        direction="row"
        align="center"
        border={{ color: 'brand', size: 'large' }}
        pad="medium"
      >
        <Button label="increase" onClick={increase} />

        <Stack anchor="bottom-right">
          <Notification size="large" />
          <Box background="brand" pad={{ horizontal: 'xsmall' }} round>
            <Text>{count}</Text>
          </Box>
        </Stack>

        <Rotate>
          <Chrome color="brand"></Chrome>
        </Rotate>
      </Box>
    </>
  );
};

export default Grommet;
