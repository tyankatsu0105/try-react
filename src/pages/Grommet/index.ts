import { lazy } from 'react';

export const Grommet = lazy(() => import('./Grommet'));
