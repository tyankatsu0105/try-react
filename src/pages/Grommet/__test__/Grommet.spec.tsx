import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';

import Grommet from '../Grommet';

describe('Grommet', () => {
  it('snapshot', () => {
    const { baseElement } = render(
      <MemoryRouter initialEntries={['/']}>
        <Grommet />
      </MemoryRouter>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
