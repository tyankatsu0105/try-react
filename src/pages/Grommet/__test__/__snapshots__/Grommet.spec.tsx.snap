// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grommet snapshot 1`] = `
<body>
  <div>
    <div
      class="StyledBox-sc-13pk1d4-0 eIHjGL"
    >
      <button
        class="StyledButton-sc-323bzc-0 kHrfdt"
        type="button"
      >
        increase
      </button>
      <div
        class="StyledStack-ajspsk-0 cBSAEZ"
      >
        <div
          class="StyledStack__StyledStackLayer-ajspsk-1 zZZtW"
        >
          <svg
            aria-label="Notification"
            class="StyledIcon-ofa7kd-0 TfAtP"
            viewBox="0 0 24 24"
          >
            <path
              d="M4,19 L4,9 C4,4.582 7.582,1 12,1 C16.418,1 20,4.582 20,9 L20,19 M1,19 L23,19 M15,19 L15,20 C15,21.657 13.657,23 12,23 C10.343,23 9,21.657 9,20 L9,19"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </div>
        <div
          class="StyledStack__StyledStackLayer-ajspsk-1 iixykT"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 dRDAJM"
          >
            <span
              class="StyledText-sc-1sadyjn-0 fWSbXS"
            >
              0
            </span>
          </div>
        </div>
      </div>
      <div
        class="css-11a3fsk"
      >
        <svg
          aria-label="Chrome"
          class="StyledIcon-ofa7kd-0 hZFnkc"
          viewBox="0 0 24 24"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <path
              d="M11.9733005,16.4144516 C9.59429509,16.4144516 7.65884342,14.4788121 7.65884342,12.0999945 C7.65884342,9.72075429 9.59429509,7.78530262 11.9733005,7.78530262 C14.3523059,7.78530262 16.2879454,9.72075429 16.2879454,12.0999945 C16.2879924,14.4788121 14.3523529,16.4144516 11.9733005,16.4144516 L11.9733005,16.4144516 Z"
              fill="#4285F4"
            />
            <path
              d="M13.7910066,17.1810894 C13.1872494,17.4007528 12.5549364,17.5116882 11.9068426,17.5116882 C10.6352186,17.5116882 9.39370022,17.0652225 8.41129728,16.2548117 C7.61431896,15.5972306 7.02033082,14.7318218 6.69297277,13.7500294 L6.691235,13.7442055 L1.93641793,5.50854377 C0.228238232,8.1357683 -0.377867323,11.2684017 0.230163868,14.3493248 C0.849467072,17.4870306 2.65331799,20.1955546 5.30970885,21.9764857 C6.69353637,22.9040785 8.22657716,23.5227711 9.86735961,23.8160783 L13.7910066,17.1810894 L13.7910066,17.1810894 Z"
              fill="#4AAE48"
            />
            <path
              d="M22.7599128,6.70666487 C19.7903479,0.731840215 12.5393434,-1.70437065 6.56465968,1.26519432 C5.01832731,2.03375777 3.65958002,3.12347966 2.57441389,4.45616042 L6.59105498,11.413435 C6.85172029,9.39264477 8.25048322,7.60626547 10.3219975,6.95469613 C10.8217704,6.79763941 11.341739,6.71403864 11.865371,6.70666487 L22.7599128,6.70666487 Z"
              fill="#EA3939"
            />
            <path
              d="M11.9264747,24 C14.936431,24 17.8171819,22.8712018 20.0368292,20.8218558 C22.2681243,18.7619893 23.6231612,15.9588274 23.8523118,12.9290041 C23.983349,11.1937716 23.7261592,9.41711443 23.1082181,7.78530262 L15.2520944,7.78530262 C16.5738788,8.83162726 17.3494403,10.4306093 17.3416908,12.1250278 C17.3360548,13.3449884 16.9177692,14.5439079 16.1598672,15.5064909 L11.1518525,23.9751076 C11.4097938,23.9916399 11.6696606,24 11.926052,24 L11.9264747,24 L11.9264747,24 Z"
              fill="#FED14B"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</body>
`;
